src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2565,10 +2733,17 @@ public override BoundNode VisitAssignmentOperator(BoundAssignmentOperator node)
             }
             else
             {
-                ReportAssignmentWarnings(node.Right, left.Type, right.Type, useLegacyWarnings: UseLegacyWarnings(node.Left));
-                TrackNullableStateForAssignment(node.Right, left.Type, left.Slot, right.Type, right.Slot);
+                var leftType = leftResult.Type;
+                var rightType = ApplyConversion(right, right, conversion, leftType.TypeSymbol, rightResult.Type, checkConversion: true, fromExplicitCast: false, out bool canConvert);
+                // Need to report all warnings that apply since the warnings can be suppressed individually.
+                ReportNullReferenceAssignmentIfNecessary(right, leftType, rightType, UseLegacyWarnings(left));
+                if (conversion.Exists && !canConvert)
+                {
+                    ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_NullabilityMismatchInAssignment, right.Syntax, GetTypeAsDiagnosticArgument(rightResult.Type?.TypeSymbol), leftType.TypeSymbol);
+                }
+                TrackNullableStateForAssignment(right, leftType, leftResult.Slot, rightType, rightResult.Slot);
                 // PROTOTYPE(NullableReferenceTypes): Check node.Type.IsErrorType() instead?
-                _result = node.HasErrors ? Result.Create(TypeSymbolWithAnnotations.Create(node.Type)) : left;
+                _result = node.HasErrors ? Result.Create(TypeSymbolWithAnnotations.Create(node.Type)) : leftResult;
```

Comment:
>leftResult [](start = 104, length = 10)

shouldn't the result take the nullability from the right?

Reply:
Good catch, thanks. Fixed and added test.

---
In reply to: [190767143](https://github.com/dotnet/roslyn/pull/26356#discussion_r190767143) [](ancestors = 190767143)